lib/tasks/prune_reports.rake

```diff
@@ -50,15 +91,19 @@ UNITS:
       exit 1
     end
 
+    if condition != "!="
+      condition = conditions[condition]
+    end
+
     # Thin query for nodes with reports that may be pruned.
     # By selecting the 'id' column only, it does not eager load all of the
     # nodes and definitely not all of the reports, making this much faster.
     cutoff = Time.now.gmtime - (upto * units[unit].to_i)
     affected_nodes = Node.select('DISTINCT nodes.id') \
                          .joins('LEFT OUTER JOIN reports ON reports.node_id = nodes.id') \
-                         .where(['reports.time < ?', cutoff])
+                         .where("reports.time < \"#{cutoff}\" and reports.status #{condition} \"#{status}\"")
```

Comment:
Even though it's interpolating on `condition`, I think it's still better to use the ? syntax here:
``` ruby
.where("reports.time < ? AND reports.status #{condition} ?", cutoff, status)
```